File: leetcode/0020-valid-parentheses/solution.py
class Solution(object):
    def isValid(self, s):
        l=['a']
        d={')':'(','}':'{',']':'['}
        for i in s:
            s=len(l)
            if i in d.keys():
                if l[s-1]!=d[i]:
                    l.append(i)
                else:
                    l.pop(s-1)
            else:
                l.append(i)
            
        if l==['a']:
            return True
        else:
            return False
                
        """
        :type s: str
        :rtype: bool
        """
        

File: leetcode/0020-valid-parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stck;
        unordered_map<char,char> mp = {{')','('},{'}','{'},{']','['}};
        for (char i : s){
            if (mp.count(i)){
                if (!stck.empty() && stck.top()==mp[i]){
                    stck.pop();
                }
                else{
                    return false;
                }
            }else {
                stck.push(i);
            }
        
            }
            return stck.empty();
        }
    };

